docs/deployment/rds-s3/auto-rds-s3-setup.py

```diff
@@ -0,0 +1,743 @@
+            }
+        ]
+    )["SecurityGroups"]
+
+    def get_security_group_id(security_group):
+        return security_group["GroupId"]
+
+    return list(map(get_security_group_id, security_groups))
+
+
+def wait_for_rds_db_instance_to_become_available(rds_client):
+    status = None
+
+    print("Waiting for RDS DB instance to become available...")
+
+    while status != "available":
+
+        status = rds_client.describe_db_instances(
+            DBInstanceIdentifier=DB_INSTANCE_NAME
+        )["DBInstances"][0]["DBInstanceStatus"]
+
+        if status == "failed":
+            raise Exception("An unexpected error occurred while waiting for the RDS DB instance to become available!")
+
+        sleep(1)
+
+    print("RDS DB instance is available!")
+
+
+def create_rds_secret(
+        secrets_manager_client,
+        rds_secret_name,
+        rds_root_password
+):
+    print("Creating RDS secret...")
+
+    db_instance_info = get_db_instance_info()
+
+    secret_string = json.dumps({
+        "username": f"{db_instance_info['MasterUsername']}",
+        "password": f"{rds_root_password}",
+        "database": f"{db_instance_info['DBName']}",
+        "host": f"{db_instance_info['Endpoint']['Address']}",
+        "port": f"{db_instance_info['Endpoint']['Port']}"
+    })
+
+    secrets_manager_client.create_secret(
+        Name=rds_secret_name,
+        Description="Kubeflow RDS secret",
+        SecretString=secret_string
+    )
+
+    print("RDS secret created!")
+
+
+def get_db_instance_info():
+    rds_client = get_rds_client()
+
+    return rds_client.describe_db_instances(
+        DBInstanceIdentifier=DB_INSTANCE_NAME
+    )["DBInstances"][0]
+
+
+def setup_cluster_secrets():
+    print("=================================================================")
+    print("                    Cluster Secrets Setup")
+    print("=================================================================")
+
+    setup_iam_service_account()
+    setup_secrets_provider()
+
+
+def setup_iam_service_account():
+    create_secrets_iam_service_account()
+
+
+def create_secrets_iam_service_account():
+    print("Creating secrets IAM service account...")
+
+    subprocess.run([
+        "eksctl",
+        "create",
+        "iamserviceaccount",
+        "--name",
+        "kubeflow-secrets-manager-sa",
+        "--namespace",
+        "kubeflow",
+        "--cluster",
+        CLUSTER_NAME,
+        "--attach-policy-arn",
+        "arn:aws:iam::aws:policy/AmazonSSMReadOnlyAccess",
+        "--attach-policy-arn",
+        "arn:aws:iam::aws:policy/SecretsManagerReadWrite",
+        "--approve",
+        "--override-existing-serviceaccounts",
+        "--region",
+        CLUSTER_REGION
+    ])
+
+    print("Secrets IAM service account created!")
+
+
+def setup_secrets_provider():
+    print("Installing secrets provider...")
+    install_secrets_store_csi_driver()
+    print("Secrets provider install done!")
+
+
+def install_secrets_store_csi_driver():
```

Comment:
Are these the only files necessary for driver installations. ? If you're following any docs for this, please add it here in comment.

Reply:
Yes these come from the manual steps (see step #4) from https://github.com/awslabs/kubeflow-manifests/tree/v1.3-branch/distributions/aws/examples/rds-s3#22-option-2-manual-setup